proc/cut.go

```diff
@@ -27,96 +34,163 @@ type Cut struct {
 // do this now since it might confuse users who expect to see output
 // fields in the order they specified.
 func CompileCutProc(c *Context, parent Proc, node *ast.CutProc) (*Cut, error) {
-	resolvers, err := expr.CompileFieldExprs(node.Fields)
+	var fields []string
+	for _, field := range node.Fields {
+		fields = append(fields, expr.FieldExprToString(field))
+	}
+
+	_, err := expr.CompileFieldExprs(node.Fields)
 	if err != nil {
 		return nil, fmt.Errorf("compiling cut: %w", err)
 	}
-	builder, err := NewColumnBuilder(c.TypeContext, node.Fields)
+	_, err = NewColumnBuilder(c.TypeContext, fields)
```

Comment:
Can we do this only if complement is false?  And add a comment explaining that this is for the purpose of error checking?